lib/core/utils.js

```diff
@@ -2026,26 +2026,57 @@ RESUtils.toolTipTimers = {};
 Each chunk is handled asynchronously from the others with a delay betwen each batch.
-If the provided callback returns false iteration will be halted.
-*/
-RESUtils.forEachChunked = function(array, chunkSize, delay, call) {
-	if (typeof array === 'undefined' || array === null) return;
-	if (typeof chunkSize === 'undefined' || chunkSize === null || chunkSize < 1) return;
-	if (typeof delay === 'undefined' || delay === null || delay < 0) return;
-	if (typeof call === 'undefined' || call === null) return;
-	var counter = 0,
-		length = array.length;
+As it stands, nope and nuh_uh are old variables not used for the time being for backwards compatibility.
+This will change the timeout dynamically based on current screen performance in an effort
+to work as fast as possibly without blocking the screen.  There are a lot of tweeks available:
 
-	function doChunk() {
-		for (var end = Math.min(length, counter + chunkSize); counter < end; counter++) {
-			var ret = call(array[counter], counter, array);
-			if (ret === false) return;
-		}
-		if (counter < length) {
-			setTimeout(doChunk, delay);
-		}
-	}
-	setTimeout(doChunk, delay);
+size: doesn't matter much, but if a particularly easy loop is regularly reaching 100's per frame
+you could eeek out a tiny gain by starting at 100's per frame.
+
+framerate: is the important one.  We may want 60, but mostly we want our scrolling and
+mouseclicks to register in a timely manner.  24 is hardly noticable I've found,
+even as low as 8-16 is tolerable for mouse clicks while the page loads.
+
+nerf: is trickey.  It basically keeps things behaving smoothly.  If you have alist that has items that processes
+orders of magnitude slower or faster, you wan't a lower nerf value.  Very smooth and symetrical loading could Ideally be at 1.
+Of course the items' process times don't matter nearly as much as how hard the browser is working on other stuff at the moment.
+*/
+RESUtils.forEachChunked = function(array, nope, nuh_uh, process, options){
+	if (typeof process !== 'function') return;
+	var	arr = Array.from(array),
+		start,
+		stop,
+		fr,
+		started = false,
+		opt = Object.assign({
+			size: 25, //starting size
+			framerate: 30, //target framerate
+			nerf: 0.9, //Be careful with this one
+		}, options),
+		size = opt.size,
+		nerf = opt.nerf,
+		framerate = opt.framerate,
+		now = function(){ return window.performance.now(); },
+		again = (typeof window.requestAnimationFrame == 'function')?
+			function(callback){ window.requestAnimationFrame(callback); }:
+			function(callback){ setTimeout(callback, 1000/opt.framerate); },
+		optimize = function(){
+			stop = now();
+			fr = 1000/(stop - start);
+			size = Math.ceil(size*(1 + (fr/framerate - 1)*nerf));
+			return (start = stop);
+		};
+	return new Promise(function(resolve, reject){
+		var doChunk = function(){
+			if (started) optimize();
+			else started = true;
+			arr.splice(0, size).forEach(process);
```

Comment:
I don't see the benefit to having a pre-calculated chunk size and trying to adjust it after - why not keep processing items until you run out of time?
Something like:
```js
const end = now() + frameTime; // frameTime = 1000 / framerate * dutyCycle
for (let i = 0; i < arr.length; ++i) {
	if (now() > end) return nextChunk(arr.slice(i));
	callback(arr[i], i, arr);
}
```

This obviates the need to guess the initial chunk size, allows it to react instantly to changes in browser performance, and guarantees that each chunk will be the optimal size (+/- 1).

This also happens to be the strategy used by FastDOM (might be a FastDOM plugin now), and I don't see any notable disadvantage over the current implementation, so I have reason to believe it's at least marginally better.

We could probably even combine the two strategies and guess if the next item will take too long, i.e.

```js
const start = now();
for (let i = 0; i < arr.length; ++i) {
	if (i && (now() - start) * (1 + 1 / i) > frameTime) return nextChunk(arr.slice(i));
	callback(arr[i], i, arr);
}
```